### DISCUSSION TITLE
Is there something planned to handle text translations (snippets)?

### DISCUSSION THREAD

**Discussion by benjamin-ott** [shopware/frontends#24] [Q&A] [answered]
Hi!
The pwa has some mechanisms to handle snippet imports and stuff like that. Is there something similar planned for the shopware frontends or is this something each individual project has to handle itself?
Thank you!

**Comment by elkmod** [comment, marked as answer]
Hi @benjamin-ott,

for now, we do not have a specific goal in terms of snippet handling. Snippet management is usually handled at build time and we've noticed that in Shopware-PWA there were wildly varying expectations/opinions about how it should work. For that reason, we've decided to leave it out of the framework, at least for now.

We might implement snippet translations into the demo-store-template at some point, to create a reference that people can base their implementations on.

**Reply by benjamin-ott** [reply]
Hi @elkmod,
thank you for the infos, that helps with our planning.

**Reply by mrsunshine** [reply]
In my opinion in would be great to have an i18n on place from the beginning. It was the first thing which i realised when i checked the code. If there is no i18n used, we have to touch every template file when starting a new project. Would be much nicer if we only had to adjust some labels in a language file or add for example a new german translation.

**Reply by StanislovasRamanauskas** [reply]
> In my opinion in would be great to have an i18n on place from the beginning. It was the first thing which i realised when i checked the code. If there is no i18n used, we have to touch every template file when starting a new project. Would be much nicer if we only had to adjust some labels in a language file or add for example a new german translation.

Agree with @mrsunshine. Absence of i18n was one of the cons for us, when deciding whether to use the project.

**Comment by BrocksiNet** [comment]
Hi all,

we have a [POC implementation via a language switcher](https://stackblitz.com/github/mkucmus/language-translations?file=app.vue). You can see the code here on StackBlitz to get an idea how to solve this task.

So it is already possible to have everything translated. We will discuss about integrating this into the demo store, as it seems to be a often requested feature.

:wave: bjoern

**Comment by benjamin-ott** [comment]
Hi, @BrocksiNet 
Thanks for the example. I tried this in our project and the translation works for text snippets.

But in the end we need a domain structure like this:

![Bildschirmfoto 2023-03-20 um 09 02 40](https://user-images.githubusercontent.com/8365682/226280683-1eaf312a-1401-4d60-b6b7-5e6cbe57e9ab.png)

I'm trying to find out if i configured something wrong, or if this is just not possible yet. 
It seems there is no mechanism right now for this (like the old pwa had...) Am i right?

**Reply by BrocksiNet** [reply]
You are right. We have some discussion ongoing also with core for that topic. At the moment working on different use cases to discuss and then in the future we will have some more flexible solution for headless frontends. Feel free to describe your preferred solution so we can consider this too. Have a great week. :hugs:

**Reply by benjamin-ott** [reply]
Ok, so at the moment it is not possible to have a headless multi lingual shop running, or are there any other possibilities I'm not aware of?

**Reply by BrocksiNet** [reply]
It's possible, but the headless frontend must run on the domains you added in the admin panel (one instance different locations guess can be achived via server config). Or you need a static mapping to send different domains (depending on the language you choose) to the request(s) for which the storefront URL is needed. Because if you don't send the correct storefront URL, the admin won't know in which language to send an email to the customer, and won't know where to link to from emails. So this part need to be reworked and we need more clear documentation here.
